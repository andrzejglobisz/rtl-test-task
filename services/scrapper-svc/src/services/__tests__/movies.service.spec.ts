import * as moxios from 'moxios';
import * as sinon from 'sinon';

import { MoviesService } from '../movies.service';
import { moviesMock, moviesFromApiMock } from '../../__mocks__/movies.mock';
import { castFromApiMock } from '../../__mocks__/cast.mock';

import axiosInstance from '../../utils/axios-instance.util';

const sandbox = sinon.createSandbox();

describe('Movies service', () => {
    const moviesService = new MoviesService();
    beforeEach(() => {
        moxios.install();
    });

    afterEach(() => {
        moxios.uninstall();
        sandbox.restore();
    });

    it('should call for cast list and save movies', () => {
        moxios.withMock(async () => {
            sandbox.stub(axiosInstance, 'get').returns({ data: [...castFromApiMock] });
            moviesService.dbService.saveMovie = jest.fn();

            const returnedMovieWithCast = await moviesService.getAndSaveCast(moviesMock[0]);

            expect(moviesService.dbService.saveMovie).toHaveBeenCalledWith(returnedMovieWithCast);
            expect(returnedMovieWithCast).toMatchSnapshot();
        });
    });

    it('should call for movies', () => {
        moxios.withMock(async () => {
            sandbox.stub(axiosInstance, 'get').returns({ data: [...moviesFromApiMock] });
            moviesService.promiseRetryWrapper = jest.fn();

            await moviesService.getMovies();

            expect(moviesService.promiseRetryWrapper).toHaveBeenCalledTimes(moviesFromApiMock.length);
        });
    });
});
