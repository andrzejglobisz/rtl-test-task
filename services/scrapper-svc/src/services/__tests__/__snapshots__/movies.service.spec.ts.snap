// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies service should call for cast list and save movies 1`] = `
Object {
  "cast": Array [
    Object {
      "birthday": "1963-04-08",
      "id": 9,
      "name": "Dean Norris",
    },
    Object {
      "birthday": "1978-01-15",
      "id": 4,
      "name": "Eddie Cahill",
    },
    Object {
      "birthday": "1979-02-01",
      "id": 2,
      "name": "Rachelle Lefevre",
    },
    Object {
      "birthday": "1979-07-17",
      "id": 1,
      "name": "Mike Vogel",
    },
    Object {
      "birthday": "1988-02-24",
      "id": 3,
      "name": "Alex Koch",
    },
    Object {
      "birthday": "1996-09-12",
      "id": 5,
      "name": "Colin Ford",
    },
    Object {
      "birthday": "1996-11-22",
      "id": 7,
      "name": "Mackenzie Lintz",
    },
  ],
  "id": 1,
  "name": "Under the Dome",
}
`;
